File: lib/data/non/initializer.rb
class NON
    require "non/app"
    require "non/audio"
    require "non/files"
    require "non/graphics"
    require "non/lights"
    require "non/network"
    require "non/particles"
    require "non/physics"
    require "non/shapes"
    require "non/input/accelerometer"
    require "non/input/keyboard"
    require "non/input/mouse"
    require "non/input/touch"
    
    @audio = Audio.new
    @files = Files.new
    @graphics = Graphics.new
    @physics = Physics.new
    @particles = Particles.new
    @lights = Lights.new
    @network = Network.new
    @shapes = Shapes.new
    @accelerometer = Accelerometer.new
    @keyboard = Keyboard.new
    @mouse = Mouse.new
    @touch = Touch.new
    
    class << self
        attr_reader :audio, :files, :graphics, :physics,
                    :particles, :lights, :network, :shapes, 
                    :accelerometer, :keyboard, :mouse, :touch
        include App
    end
end

# Callbacks
def init(assets); end
def ready; end
def update(dt); end
def draw; end
def close; end
def pause; end
def resume; end
def resize(w, h); end

# Input Callbacks
def keydown(key); end
def keyup(key); end
def keytyped(character); end
def touchdown(x, y, pointer, button); end
def touchup(x, y, pointer, button); end
def touchdragged(x, y, pointer); end
def mousemoved(x, y); end
def mousescrolled(amount); end

# Network Callbacks
def connected(connection); end
def disconnected(connection); end
def received(connection, data); end

require 'main'

File: lib/data/non/lua/initializer.lua
-- reserved globals - NON_NON, NON_GDX, NON_MODULE, NON_SCRIPT

non = {}

function require(file)
  return NON_SCRIPT:eval(NON_GDX.files:internal(file..".lua"):readString())
end

require "non/lua/files"
require "non/lua/app"
require "non/lua/audio"
require "non/lua/graphics"
require "non/lua/network"
require "non/lua/input/accelerometer"
require "non/lua/input/keyboard"
require "non/lua/input/mouse"
require "non/lua/input/touch"

-- Callbacks
function non.ready() end
function non.update(dt) end
function non.draw() end
function non.quit() end
function non.pause() end
function non.resume() end
function non.resize(w, h) end

-- Input Callbacks
function non.keydown(key) end
function non.keyup(key) end
function non.keytyped(character) end
function non.touchdown(x, y, pointer, button) end
function non.touchup(x, y, pointer, button) end
function non.touchdragged(x, y, pointer) end
function non.mousemoved(x, y) end
function non.mousescrolled(amount) end

-- Network Callbacks
function non.connected(connection) end
function non.disconnected(connection) end
function non.received(connection, data) end

require "main"